src/main/java/org/jabref/gui/entryeditor/LatexReferencesTabViewModel.java

```diff
@@ -0,0 +1,102 @@
+import javafx.beans.property.ReadOnlyListWrapper;
+import javafx.beans.property.SimpleBooleanProperty;
+import javafx.beans.property.SimpleObjectProperty;
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+
+import org.jabref.gui.AbstractViewModel;
+import org.jabref.gui.util.BackgroundTask;
+import org.jabref.gui.util.TaskExecutor;
+import org.jabref.logic.texparser.DefaultTexParser;
+import org.jabref.model.database.BibDatabaseContext;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.texparser.Citation;
+import org.jabref.model.texparser.TexParserResult;
+import org.jabref.preferences.PreferencesService;
+
+public class LatexReferencesTabViewModel extends AbstractViewModel {
+
+    private static final String TEX_EXT = ".tex";
+    private final BibDatabaseContext databaseContext;
+    private final PreferencesService preferencesService;
+    private final TaskExecutor taskExecutor;
+    private final ObjectProperty<BibEntry> entry;
+    private final ObservableList<Citation> citationList;
+    private final BooleanProperty searchInProgress;
+    private final BooleanProperty successfulSearch;
+
+    public LatexReferencesTabViewModel(BibDatabaseContext databaseContext, PreferencesService preferencesService,
+                                       TaskExecutor taskExecutor) {
+        this.databaseContext = databaseContext;
+        this.preferencesService = preferencesService;
+        this.taskExecutor = taskExecutor;
+        this.entry = new SimpleObjectProperty<>();
+        this.citationList = FXCollections.observableArrayList();
+        this.searchInProgress = new SimpleBooleanProperty(true);
+        this.successfulSearch = new SimpleBooleanProperty(false);
+    }
+
+    public void setEntry(BibEntry entry) {
+        this.entry.set(entry);
+    }
+
+    public ObservableList<Citation> getCitationList() {
+        return new ReadOnlyListWrapper<>(citationList);
+    }
+
+    public BooleanProperty searchInProgressProperty() {
+        return searchInProgress;
+    }
+
+    public BooleanProperty successfulSearchProperty() {
+        return successfulSearch;
+    }
+
+    public void initSearch() {
+        BackgroundTask.wrap(this::searchAndParse)
+                      .onRunning(() -> {
+                          searchInProgress.set(true);
+                          successfulSearch.set(false);
+                      })
+                      .onSuccess(resultsFound -> {
+                          successfulSearch.set(resultsFound);
+                          searchInProgress.set(false);
+                      })
+                      .executeWith(taskExecutor);
+    }
+
+    private boolean searchAndParse() throws IOException {
+        Path directory = databaseContext.getMetaData().getLaTexFileDirectory(preferencesService.getUser())
+                                        .orElse(preferencesService.getWorkingDir());
+
+        List<Path> texFiles;
```

Comment:
Possible Null when the stream breaks with an exception?

Reply:
Yes. When it is `null`, the progress indicator is displayed forever. I have thought to show an error message inside of the entry editor tab (without dialog boxes), simply by adding the `onFailure()` event in `initSearch()`. Do you agree?

Reply:
yes, showing an error message without an extra dialog is good. in the entry editor we should avoid modal dialogs

Reply:
Or you init it as new ArrayList<> and check if it's empty